### ISSUE TITLE
Profiles with inconsistent values for previously existing Policies

### ISSUE THREAD

**Issue by bmg13** [eclipse-tractusx/tractusx-edc#1705]
## Describe the bug

When introducing profiles in Policies, [the respective schema was updated ](https://github.com/eclipse-tractusx/tractusx-edc/blob/2be7d26fe8718c27121517f34b69024f40f905de/edc-extensions/migrations/control-plane-migration/src/main/resources/org/eclipse/tractusx/edc/postgresql/migration/policy/V0_0_5__Add_Profiles.sql#L16-L15)accordingly. However, this did not take under consideration the fact that pre-existing policies would have a `null` value for this column instead of an empty collection.
So, while querying the DB for policies, a `NPE` is launched, breaking its use. 

### To Reproduce

Deploying a new instance in an infrastructure containing (stored) policies built before the addition of the `profiles` property. Simply retrieve (GET all, for example) policies and a `500` HTTP error will likely be seen.

### Expected behavior

Backwards compatible with previous structure. Aim could be to have a default `profiles` value of an empty collection. Avoiding a null would be needed.

### Screenshots/Error Messages

<img width="492" alt="Screenshot 2024-12-03 at 22 52 35" src="https://github.com/user-attachments/assets/ef952543-c6ce-4024-9e2a-dd828a54505b">

<img width="530" alt="Screenshot 2024-12-03 at 23 13 20" src="https://github.com/user-attachments/assets/4b4a5719-07f1-4531-83f1-67e48000c3dc">

- Used version: 0.8.0 of Tractus-X

## Possible Implementation

An implementation with a couple of steps may solve this.
First, create an update values in table `edc_policydefinitions` in migrations to ensure not null existing values, similar to:

```
UPDATE edc_policydefinitions SET profiles='[]'::json where profiles is NULL;
```

Also, ensure programmatically that this condition happens, in order to avoid the `NPE`. A possible suggestion would be on the Upstream repo, having a validation before calling the `Policy.Builder`.

**Comment by ndr-brt** [comment]
this was my fault, definitely, anyway curious that this has been spotted only now, the migration has been there for more than 4 months.
We could add a note in the `0.8.0` release and in the migration guide as a suggestion to workaround this problem, then provide the fix for the `0.8.1` version.

**Comment by rafaelmag110** [comment]
I guess this (as with most of other more trivial issues that should had been caught already) happens because adopters only start deploying after the version is officially released. I.e., no one keeps a deployed version of the main branch of tractusx-edc...
